api/app/controllers/spree/api/payments_controller.rb

```diff
@@ -17,8 +17,10 @@ def new
       end
 
       def create
-        @payment = @order.payments.build(payment_params)
-        if @payment.save
+        success = @order.contents.update_cart(payments_attributes: [payment_params])
+        @payment = @order.payments.last
+
+        if success
           respond_with(@payment, status: 201, default_template: :show)
         else
           invalid_resource!(@payment)
```

Comment:
Yeah I was worried about that also but it's cached inside the association since we just tried to add it through the association.  There's a spec that verifies this: https://github.com/solidusio/solidus/blob/458efae129fe3a4bec1aed105f47935915f5a0f2/api/spec/controllers/spree/api/payments_controller_spec.rb#L54